test/e2e/networking.go

```diff
@@ -101,153 +102,223 @@ var _ = Describe("Networking", func() {
 		}
 	})
 
-	//Now we can proceed with the test.
-	It("should function for intra-pod communication", func() {
+	//Each tuple defined in this struct array represents
+	//a number of services, and a timeout.  So for example,
+	//{1, 300} defines a test where 1 service is created and
+	//we give it 300 seconds to timeout.  This is how we test
+	//services in parallel... we can create a tuple like {5,400}
+	//to confirm that services over 5 ports all pass the networking test.
+	serviceSoakTests := []struct {
+		service        int
+		timeoutSeconds time.Duration
+	}{
+		//These are very liberal, once this test is running regularly,
+		//We will DECREASE the timeout value.
+		//Replace this with --scale constants eventually https://github.com/kubernetes/kubernetes/issues/10479.
+		{1, time.Duration(100 * time.Second)},
+		{3, time.Duration(200 * time.Second)}, //Test that parallel nettests running on different ports complete.
+	}
 
-		By(fmt.Sprintf("Creating a service named %q in namespace %q", svcname, f.Namespace.Name))
-		svc, err := f.Client.Services(f.Namespace.Name).Create(&api.Service{
-			ObjectMeta: api.ObjectMeta{
-				Name: svcname,
-				Labels: map[string]string{
-					"name": svcname,
-				},
-			},
-			Spec: api.ServiceSpec{
-				Ports: []api.ServicePort{{
-					Protocol:   "TCP",
-					Port:       8080,
-					TargetPort: util.NewIntOrStringFromInt(8080),
-				}},
-				Selector: map[string]string{
-					"name": svcname,
-				},
-			},
-		})
-		if err != nil {
-			Failf("unable to create test service named [%s] %v", svc.Name, err)
-		}
+	// TODO: support DNS on vagrant #3580
+	for _, svcSoak := range serviceSoakTests {
+		//copy to local to avoid range overwriting
+		timeoutSeconds := svcSoak.timeoutSeconds
+		serviceNum := svcSoak.service
+		It(fmt.Sprintf("should function for intrapod communication between all hosts in %v parallel services", serviceNum),
+			func() {
+				SkipIfProviderIs("vagrant")
```

Comment:
You do not need to change that in this PR, I will make a note to no longer need it next time I run conformance tests.